ministryofjustice/laa-apply-for-criminal-legal-aid
# [CRIMAPP-735] Refactor display absent records

## Description of change
Refactors the following sections to display based on newly added attributes: income payments, income benefits, outgoings payments, savings, investments and national savings certificates

This will make the display of the summary card more accurate as not all applications go through the sections/are shown the questions

## Link to relevant ticket
Technically relates to [CRIMAPP-697](https://dsdmoj.atlassian.net/browse/CRIMAPP-697) but i did it as part of [CRIMAPP-735](https://dsdmoj.atlassian.net/browse/CRIMAPP-735)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

No changes to display

## How to manually test the feature
Requires datastore schema changes to test locally: [#245](https://github.com/ministryofjustice/laa-criminal-applications-datastore/pull/245)

Check the display of the following means assessment sections:  income payments, income benefits, outgoings payments, savings, investments and national savings certificates

Confirm the correct display when: there are records, there are no records (because no records was selected), there are no records (because the question was not shown, in the case of the capital sections)


[CRIMAPP-697]: https://dsdmoj.atlassian.net/browse/CRIMAPP-697?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMAPP-735]: https://dsdmoj.atlassian.net/browse/CRIMAPP-735?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
